### PR TITLE
[FIX] crm: handle customer creation on. lead -> oppor. w/ archived lang

### PR DESCRIPTION
**Pull request by xavieralt**
To reproduce:

1. enable 'debug' mode
2. go to menu "Settings / Translations / Languages" and enable a new language (ex: nl_NL) 3 create a new lead and set that language
  (IMPORTANT: do not set a customer)
4. go back to menu "Settings / Translations / Languages" and disable the language you previously enabled at step 2.
5. from the lead list view, selection the lead you created at step 3, click "Action" and choose "Convert to opportunities"
6. In the wizard choose for:
   * Conversion Action: Convert to opportunity
   * Customer: Create a new customer
7. Click button "Convert to opportunity"

This crash with the following error:

```
Traceback (most recent call last):
  ...

  File "/data/build/odoo/addons/crm/wizard/crm_lead_to_opportunity_mass.py", line 103, in action_mass_convert
    return self.action_apply()
  File "/data/build/odoo/addons/crm/wizard/crm_lead_to_opportunity.py", line 121, in action_apply
    result_opportunity = self._action_convert()
  File "/data/build/odoo/addons/crm/wizard/crm_lead_to_opportunity.py", line 144, in _action_convert
    self._convert_and_allocate(result_opportunities, [self.user_id.id], team_id=self.team_id.id)
  File "/data/build/odoo/addons/crm/wizard/crm_lead_to_opportunity_mass.py", line 78, in _convert_and_allocate
    return super(Lead2OpportunityMassConvert, self)._convert_and_allocate(leads, salesmen_ids, team_id=team_id)
  File "/data/build/odoo/addons/crm/wizard/crm_lead_to_opportunity.py", line 152, in _convert_and_allocate
    self._convert_handle_partner(
  File "/data/build/odoo/addons/crm/wizard/crm_lead_to_opportunity_mass.py", line 109, in _convert_handle_partner
    return super(Lead2OpportunityMassConvert, self)._convert_handle_partner(lead, action, partner_id)
  File "/data/build/odoo/addons/crm/wizard/crm_lead_to_opportunity.py", line 166, in _convert_handle_partner
    lead.with_context(default_user_id=self.user_id.id)._handle_partner_assignment(
  File "/data/build/odoo/addons/crm/models/crm_lead.py", line 1556, in _handle_partner_assignment
    partner = lead._create_customer()
  File "/data/build/odoo/addons/crm/models/crm_lead.py", line 1692, in _create_customer
    return Partner.create(self._prepare_customer_values(self.name, is_company=False))
  File "<decorator-gen-1007>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 395, in _model_create_single
    return create(self, arg)
  File "/data/build/odoo/addons/l10n_cl/models/res_partner.py", line 55, in create
    return super().create(values)
  File "<decorator-gen-329>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 417, in _model_create_multi
    return create(self, [arg])
  File "/data/build/odoo/addons/base_vat/models/res_partner.py", line 718, in create
    return super(ResPartner, self).create(vals_list)
  File "<decorator-gen-263>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 418, in _model_create_multi
    return create(self, arg)
  File "/data/build/odoo/addons/account/models/partner.py", line 557, in create
    return super().create(vals_list)
  File "<decorator-gen-227>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 418, in _model_create_multi
    return create(self, arg)
  File "/data/build/odoo/addons/partner_autocomplete/models/res_partner.py", line 163, in create
    partners = super(ResPartner, self).create(vals_list)
  File "<decorator-gen-223>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 398, in _model_create_single
    return self.browse().concat(*(create(self, vals) for vals in arg))
  File "/data/build/odoo/odoo/api.py", line 398, in <genexpr>
    return self.browse().concat(*(create(self, vals) for vals in arg))
  File "/data/build/odoo/addons/mail_plugin/models/res_partner.py", line 32, in create
    partner = super(ResPartner, self).create(vals)
  File "<decorator-gen-100>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 417, in _model_create_multi
    return create(self, [arg])
  File "/data/build/odoo/odoo/addons/base/models/res_partner.py", line 652, in create
    partners = super(Partner, self).create(vals_list)
  File "<decorator-gen-137>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 418, in _model_create_multi
    return create(self, arg)
  File "/data/build/odoo/addons/mail/models/mail_thread.py", line 265, in create
    threads = super(MailThread, self).create(vals_list)
  File "<decorator-gen-306>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 418, in _model_create_multi
    return create(self, arg)
  File "/data/build/odoo/addons/website/models/mixins.py", line 211, in create
    records = super(WebsitePublishedMixin, self).create(vals_list)
  File "<decorator-gen-69>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 418, in _model_create_multi
    return create(self, arg)
  File "/data/build/odoo/odoo/addons/base/models/ir_fields.py", line 613, in create
    recs = super().create(vals_list)
  File "<decorator-gen-15>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 418, in _model_create_multi
    return create(self, arg)
  File "/data/build/odoo/odoo/models.py", line 4088, in create
    records = self._create(data_list)
  File "/data/build/odoo/odoo/models.py", line 4152, in _create
    col_val = field.convert_to_column(val, self, stored)
  File "/data/build/odoo/odoo/fields.py", line 2538, in convert_to_column
    value = self.convert_to_cache(value, record)
  File "/data/build/odoo/odoo/fields.py", line 2550, in convert_to_cache
    raise ValueError("Wrong value for %s: %r" % (self, value))
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/data/build/odoo/odoo/http.py", line 654, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/data/build/odoo/odoo/http.py", line 301, in _handle_exception
    raise exception.with_traceback(None) from new_cause
ValueError: Wrong value for res.partner.lang: 'nl_NL'
```

This commit ensure that when converting a lead to an opportunity, we
don't crash if language is archived.

OPW-3617359

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by reth-odoo — COMMENTED** [review]
Hey :wave: 

I think a simpler, shorter solution would be fine here. Doesn't seem like a very common issue.

File: addons/crm/tests/test_crm_lead_convert.py [hunk 1]
```diff
@@ -241,6 +242,28 @@ def test_lead_convert_corner_cases_matching(self):
         # TDE FIXME: should take into account normalized email version, not encoded one
         # self.assertEqual(convert.partner_id, customer)
 
+    @users('user_sales_manager')
+    def test_lead_convert_corner_cases_archived_lang(self):
+        """ Test Lead._create_customer() """
```
Comment (reth-odoo) [inline addons/crm/tests/test_crm_lead_convert.py:247]: would rather go something like `test_lead_convert_no_lang(self)` and then specify what we do in the docstring like `""" Ensure converting a lead with an archived language correctly falls back on the default partner language. """`

This way the test is more 'use case driven' than technically driven, if that makes sense.

**Review by reth-odoo — COMMENTED** [review]
Just some wording nitpick comments otherwise seems fine

In the commit message, I'd just add a short summary like "Converting a lead to an opportunity where the language was archived crashed." this way if we don't care about the exact steps it's nice and short. And I'd say the traceback isn't necessary, it's prone to going over the recommended line length of commits and in the end you don't gain much reading it in the commit.

imo it's best reserved for the task or the PR message. Matters of taste though I guess ^^

thanks for the fix!

**Comment by reth-odoo** [comment]
@jeh-odoo Hi :wave: 

There's a little one-line edge case to check here if you have some time

File: addons/crm/tests/test_crm_lead_convert.py [hunk 2]
```diff
@@ -69,6 +69,7 @@ def setUpClass(cls):
         date = Datetime.from_string('2020-01-20 16:00:00')
         cls.crm_lead_dt_mock.now.return_value = date
 
+
```
Comment (jeh-odoo) [inline addons/crm/tests/test_crm_lead_convert.py:72]: Small unnecessary extra line :eyes: :slightly_smiling_face:

**Review by jeh-odoo — COMMENTED** [review]
Hello @xavieralt 
Just an additional picky comment but otherwise code LGTM. Thanks for the fix!
I agree that the traceback in the commit message is a bit overkill :slightly_smiling_face:
Cheers!
cc @reth-odoo

File: addons/crm/tests/test_crm_lead_convert.py (hunk 1 above)
Comment (xavieralt) [reply addons/crm/tests/test_crm_lead_convert.py]: updated

File: addons/crm/models/crm_lead.py [hunk 3]
```diff
@@ -1722,7 +1722,7 @@ def _prepare_customer_values(self, partner_name, is_company=False, parent_id=Fal
             'is_company': is_company,
             'type': 'contact'
         }
-        if self.lang_id:
+        if self.lang_id and self.lang_id.active:
```
Comment (tde-banana-odoo) [inline addons/crm/models/crm_lead.py:1725]: Could be simplified directly to
```suggestion
        if self.lang_id.active:
```

File: addons/crm/tests/test_crm_lead_convert.py [hunk 4]
```diff
@@ -241,6 +241,28 @@ def test_lead_convert_corner_cases_matching(self):
...
+        lead = self.lead_1.with_user(self.env.user)
+        lead.lang_id = inactive_lang
+
+        convert = self.env['crm.lead2opportunity.partner'].with_context({
+            'active_model': 'crm.lead',
+            'active_id': self.lead_1.id,
+            'active_ids': self.lead_1.ids,
+        }).create({})
+        convert.action = "create"
```
Comment (tde-banana-odoo) [inline addons/crm/tests/test_crm_lead_convert.py:261]: could be given to "create" directly maybe ?

**Review by tde-banana-odoo — COMMENTED** [review]
Some small remarks while passing by, otherwise LGTM :)

**Comment by tde-banana-odoo** [comment]
@robodoo r+
